# Explore the Phenotype Data

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plot
import seaborn as sns

In [2]:
phenotypes = pd.read_csv('phenotype_data_merged.txt', dtype = {'rfid': str}, sep = '\t')

/var/folders/6w/pjm4r6dn0zd8yc5l111vlffh0000gn/T/ipykernel_5001/2584757680.py:1: DtypeWarning: Columns (56,110,112,146,226,299,302,303,313,419,420,480,482,563,571,607,608,610,682,683,684,697,704,772,816) have mixed types. Specify dtype option on import or set low_memory=False.
  phenotypes = pd.read_csv('phenotype_data_merged.txt', dtype = {'rfid': str}, sep = '\t')


In [3]:
phenotypes.shape

(9865, 831)

In [14]:
list(phenotypes['rfid'])[3055]

'933000120138674'

In [15]:
phenotypes.head()

rfid  dd_pcd0    dd_idpt0  dd_idpt600  dd_idpt1200  dd_idpt1800  \
0  00077E61F3    2.595  105.900000   90.041667    94.888889    69.523810   
1  00077E6207    2.180  106.838710  102.307692    96.350000    79.500000   
2  00077E6232    3.120  103.517241   73.047619    55.666667    62.700000   
3  00077E6239    3.130  118.571429   86.500000    98.333333    79.500000   
4  00077E62D2    2.440  128.439024   98.555556    78.272727    75.952381   

   dd_idpt2400  dd_idptmax  dd_idptauc  dd_tip0  ...  lr_lt_act_sa  \
0    57.444444  105.900000    0.693567     9.36  ...           NaN   
1    59.789474  106.838710    0.757020    12.64  ...           NaN   
2    63.400000  103.517241    0.553091    10.79  ...           NaN   
3    89.458333  118.571429    0.663942     7.79  ...           NaN   
4    70.000000  128.439024    0.595431     3.94  ...           NaN   

   lr_lt_inact_sa  lr_lt_tot_sa  lr_px_pre_act_sa  lr_px_pre_inact_sa  \
0             NaN           NaN               NaN                 NaN   
1             NaN           NaN               NaN                 NaN   
2             NaN           NaN               NaN                 NaN   
3             NaN           NaN               NaN                 NaN   
4             NaN           NaN               NaN                 NaN   

   lr_px_pre_tot_sa  dd_pcd600  dd_pcd1200  dd_pcd1800  dd_pcd2400  
0               NaN        NaN         NaN         NaN         NaN  
1               NaN        NaN         NaN         NaN         NaN  
2               NaN        NaN         NaN         NaN         NaN  
3               NaN        NaN         NaN         NaN         NaN  
4               NaN        NaN         NaN         NaN         NaN  

[5 rows x 831 columns]

In [16]:
len(phenotypes)

9865

There are 9865 rats in our data

### How many phenotypes and rats are left when all NaNs are dropped?

In [17]:
pheno_cleaned = phenotypes.dropna()

In [18]:
pheno_cleaned.head()

Empty DataFrame
Columns: [rfid, dd_pcd0, dd_idpt0, dd_idpt600, dd_idpt1200, dd_idpt1800, dd_idpt2400, dd_idptmax, dd_idptauc, dd_tip0, dd_tip600, dd_tip1200, dd_tip1800, dd_tip2400, dd_tipmax, dd_tipauc, dd_pc0, dd_pc600, dd_pc1200, dd_pc1800, dd_pc2400, dd_tw0, dd_tw600, dd_tw1200, dd_tw1800, dd_tw2400, dd_sess0, dd_sess600, dd_sess1200, dd_sess1800, dd_sess2400, dd_rf0, dd_rf600, dd_rf1200, dd_rf1800, dd_rf2400, dd_wrate0, dd_wrate600, dd_wrate1200, dd_wrate1800, dd_wrate2400, dd_pcrate0, dd_pcrate600, dd_pcrate1200, dd_pcrate1800, dd_pcrate2400, dd_pcratemax, dd_pcrateauc, dd_rfrate0, dd_rfrate600, dd_rfrate1200, dd_rfrate1800, dd_rfrate2400, dd_rfrateav, dd_age, cohort, dd_box_id, sex, lr_px_pre_active, lr_px_pre_inactive, lr_px_pre_total, lr_px_pre_relactive, lr_px_pre_act1_5, lr_px_pre_act4_5, lr_px_pre_act7_5, lr_px_pre_act10_5, lr_px_pre_act13_5, lr_px_pre_act16_5, lr_px_pre_act_max, lr_px_pre_inact1_5, lr_px_pre_inact4_5, lr_px_pre_inact7_5, lr_px_pre_inact10_5, lr_px_pre_inact13_5, lr_px_pre_inact16_5, lr_px_pre_inact_max, lr_px_pre_tot1_5, lr_px_pre_tot4_5, lr_px_pre_tot7_5, lr_px_pre_tot10_5, lr_px_pre_tot13_5, lr_px_pre_tot16_5, lr_px_pre_tot_max, lr_lt_active, lr_lt_inactive, lr_lt_total, lr_lt_reinforcers, lr_lt_relactive, lr_lt_act1_5, lr_lt_act4_5, lr_lt_act7_5, lr_lt_act10_5, lr_lt_act13_5, lr_lt_act16_5, lr_lt_act_max, lr_lt_inact1_5, lr_lt_inact4_5, lr_lt_inact7_5, lr_lt_inact10_5, lr_lt_inact13_5, ...]
Index: []

[0 rows x 831 columns]

In [19]:
phenotypes.columns[ np.sum(phenotypes.isnull()) == 0 ]

Index(['rfid', 'sex', 'project'], dtype='object')

Anove are the only columns not missing any data

In [20]:
list(phenotypes.columns)

['rfid',
 'dd_pcd0',
 'dd_idpt0',
 'dd_idpt600',
 'dd_idpt1200',
 'dd_idpt1800',
 'dd_idpt2400',
 'dd_idptmax',
 'dd_idptauc',
 'dd_tip0',
 'dd_tip600',
 'dd_tip1200',
 'dd_tip1800',
 'dd_tip2400',
 'dd_tipmax',
 'dd_tipauc',
 'dd_pc0',
 'dd_pc600',
 'dd_pc1200',
 'dd_pc1800',
 'dd_pc2400',
 'dd_tw0',
 'dd_tw600',
 'dd_tw1200',
 'dd_tw1800',
 'dd_tw2400',
 'dd_sess0',
 'dd_sess600',
 'dd_sess1200',
 'dd_sess1800',
 'dd_sess2400',
 'dd_rf0',
 'dd_rf600',
 'dd_rf1200',
 'dd_rf1800',
 'dd_rf2400',
 'dd_wrate0',
 'dd_wrate600',
 'dd_wrate1200',
 'dd_wrate1800',
 'dd_wrate2400',
 'dd_pcrate0',
 'dd_pcrate600',
 'dd_pcrate1200',
 'dd_pcrate1800',
 'dd_pcrate2400',
 'dd_pcratemax',
 'dd_pcrateauc',
 'dd_rfrate0',
 'dd_rfrate600',
 'dd_rfrate1200',
 'dd_rfrate1800',
 'dd_rfrate2400',
 'dd_rfrateav',
 'dd_age',
 'cohort',
 'dd_box_id',
 'sex',
 'lr_px_pre_active',
 'lr_px_pre_inactive',
 'lr_px_pre_total',
 'lr_px_pre_relactive',
 'lr_px_pre_act1_5',
 'lr_px_pre_act4_5',
 'lr_px_pre_act7_5',
 '

In [21]:
phenotypes_loco_cols = list(phenotypes.filter(regex = 'loco').columns)

In [22]:
phenotypes_loco_cols = ['rfid'] + phenotypes_loco_cols

In [23]:
phenotypes_loco = phenotypes[phenotypes_loco_cols]

In [24]:
phenotypes_loco.head()

rfid  loco_maxcent  loco_cent16_5  loco_cent13_5  loco_cent10_5  \
0  00077E61F3         171.3          171.3           59.4           83.9   
1  00077E6207         115.4           40.8          115.4           12.4   
2  00077E6232          48.1           32.9           28.0           48.1   
3  00077E6239         147.1          128.2           88.7          147.1   
4  00077E62D2          89.4            0.0            0.0            0.0   

   loco_cent7_5  loco_cent4_5  loco_cent1_5  loco_maxrear  loco_rear16_5  ...  \
0          84.8          55.0          49.7          19.0           17.0  ...   
1           7.9          80.9          45.5          26.0           10.0  ...   
2          33.1          37.2          29.0          20.0           10.0  ...   
3         118.0         127.8         109.0          25.0            0.0  ...   
4           7.4          89.4          12.0          22.0            0.0  ...   

   locomotor_fecal_boli_2  locomotor_fecal_boli_change  \
0                     NaN                          NaN   
1                     NaN                          NaN   
2                     NaN                          NaN   
3                     NaN                          NaN   
4                     NaN                          NaN   

   locomotor_fecal_boli_age_1  locomotor_fecal_boli_age_2  \
0                         NaN                         NaN   
1                         NaN                         NaN   
2                         NaN                         NaN   
3                         NaN                         NaN   
4                         NaN                         NaN   

   ccp_change_in_locomotor_activity  loco_sheet_id  loco_saact  loco_sadis  \
0                               NaN            NaN         NaN         NaN   
1                               NaN            NaN         NaN         NaN   
2                               NaN            NaN         NaN         NaN   
3                               NaN            NaN         NaN         NaN   
4                               NaN            NaN         NaN         NaN   

   loco_sarear  loco_sacent  
0          NaN          NaN  
1          NaN          NaN  
2          NaN          NaN  
3          NaN          NaN  
4          NaN          NaN  

[5 rows x 73 columns]

In [25]:
len(phenotypes_loco)

9865

In [26]:
locomotor_cleaned = phenotypes_loco.dropna()

In [27]:
len(locomotor_cleaned)

0

In [28]:
phenotypes_loco.to_csv("phenotypes_loco.txt", sep = '\t')